### PR TITLE
fix: Clean up anchor request store even if stream is already anchored

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: packages/core/src/state-management/repository.ts [hunk 1]
```diff
@@ -614,10 +614,16 @@ export class Repository {
         return false
       }
       case AnchorRequestStatusName.COMPLETED: {
-        if (anchorEvent.cid.equals(state$.tip)) {
+        const anchorCommitCID = anchorEvent.witnessCar.roots[0]
+        await this._handleAnchorCommit(state$, anchorEvent.cid, anchorEvent.witnessCar)
+        if (state$.tip.equals(anchorCommitCID)) {
```
Comment (AaronGoldman) [inline packages/core/src/state-management/repository.ts:619]: Do we have a `if anchorHistory.includes(anchorCommitCID)`?

Comment (stbrody) [reply packages/core/src/state-management/repository.ts:619]: depends on what you mean by "anchorHistory".  We can check if the anchor commit CID exists anywhere in the log of the state, but if we use that as the criteria to delete the entry from the anchorRequestStore, we risk deleting a valid entry for a newer update

File: packages/core/src/state-management/repository.ts [hunk 2]
```diff
@@ -614,10 +614,16 @@ export class Repository {
         return false
       }
       case AnchorRequestStatusName.COMPLETED: {
-        if (anchorEvent.cid.equals(state$.tip)) {
+        const anchorCommitCID = anchorEvent.witnessCar.roots[0]
+        await this._handleAnchorCommit(state$, anchorEvent.cid, anchorEvent.witnessCar)
```
Comment (stephhuynh18) [inline packages/core/src/state-management/repository.ts:618]: do we expect `anchorEvent.witnessCar.roots[0]` and  `anchorEvent.cid` to be different?

Comment (stephhuynh18) [reply packages/core/src/state-management/repository.ts:618]: AH i see thanks

**Review by stephhuynh18 — APPROVED** [review]

**Pull request merged**